src/firefly/js/visualize/XYPlotCntlr.js

```diff
@@ -72,25 +74,55 @@ export function dispatchLoadPlotData(xyPlotParams, searchRequest) {
 }
 
 /*
- * Update xy plot data
- * @param {boolean} isPlotDataReady - flags that xy plot data are available
- * @param {Number[][]} xyPlotData - an array of the number arrays with rowIdx, x, y, [error]
- * @param {Object} xyPlotParams - XY plot options (column names, etc.)
+ * Set selection to give user choice of actions on selection (zoom, filter, or select points)
  * @param {String} tblId - table id
+ * @param {Object} selection - {xMin, xMax, yMin, yMax}
  */
-export function dispatchUpdatePlotData(isPlotDataReady, xyPlotData, xyPlotParams, tblId) {
-    flux.process({type: UPDATE_PLOT_DATA, payload: {isPlotDataReady, xyPlotData, xyPlotParams, tblId}});
-}
-
 export function dispatchSetSelection(tblId, selection) {
     flux.process({type: SET_SELECTION, payload: {tblId, selection}});
 }
 
-export function dispatchSetZoom(tblId, selection) {
+/*
+ * Zoom XY plot to a given selection or reset zoom if no selection is given
+ * @param {String} tblId - table id
+ * @param {Object} selection - {xMin, xMax, yMin, yMax}
+ */
+export function dispatchZoom(tblId, selection) {
+    const {xyPlotData, xyPlotParams, decimatedUnzoomed} = flux.getState()[XYPLOT_DATA_KEY][tblId];
```

Comment:
this is a good place to use lodash 'get'; too many ways to get undefined error:
get(flux.getState(), [XYPLOT_DATA_KEY,tblId], {});